runners/spark/src/main/java/org/apache/beam/runners/spark/util/GlobalWatermarkHolder.java

```diff
@@ -100,103 +126,162 @@ public static void addAll(Map<Integer, Queue<SparkWatermarks>> sourceTimes) {
     }
   }
 
-  private static synchronized void initWatermarkCache(Long batchDuration) {
-    if (watermarkCache == null) {
-      watermarkCache =
-          CacheBuilder.newBuilder()
-              // expire watermarks every half batch duration to ensure they update in every batch.
-              .expireAfterWrite(batchDuration / 2, TimeUnit.MILLISECONDS)
-              .build(new WatermarksLoader());
-    }
+  private static boolean canBypassRemoteWatermarkFetching() {
+    return driverNodeWatermarks != null;
+  }
+
+  private static synchronized LoadingCache<String, Map<Integer, SparkWatermarks>>
+      initWatermarkCache(final Long batchDuration) {
+
+    checkState(
```

Comment:
Perhaps now that this method is not a `void` that sets `watermarkCache` we can not perform this check and rename the method to something like `createWatermarkCache`, the client of this method can and does check if `watermarkCache` is null before they invoke this to create a `watermarkCache`.